### PR TITLE
Don't update the daemonsets / the network status if there are no changes

### PR DESCRIPTION
**Pull request by fedepaol**
The networkpolicy controller is configured so that daemonsets / sriov network status are subresources of the policy. What happens is that whenever a policy is added, the controller goes into an endless loop where it updates the subresources and get notified of those changes through the primary one (the policy), and then updates again and so on.

This can be seen looking at the operator's logs (I noticed it by chance running the operator locally).

If the subresources' specs do not change from what's read from the api we don't update them, stopping the loop.

### CODE REVIEW

**Comment by fedepaol** [comment]
@zshi-redhat @pliurh PTAL, I think the operator will benefit from this.

**Comment by zshi-redhat** [comment]
@fedepaol thanks for the fix!
So the reason why nodeState is updated continuously is because nodeState.Status is kept being updated.
And for daemonSet, the reason is default field that's not rendered in operator manifests is filled by k8s controllers automatically.

looks good to me.

**Comment by fedepaol** [comment]
> @fedepaol thanks for the fix!
> So the reason why nodeState is updated continuously is because nodeState.Status is kept being updated.
> And for daemonSet, the reason is default field that's not rendered in operator manifests is filled by k8s controllers automatically.
> 

Yes, I noticed it by chance, while I was testing my changes and it wasn't stopping logging

File: controllers/sriovnetworknodepolicy_controller.go [hunk 1]
```diff
@@ -287,13 +290,17 @@ func (r *SriovNetworkNodePolicyReconciler) syncSriovNetworkNodeState(np *sriovne
...
-				p.Apply(found, bool(ppp == p.Spec.Priority))
+				p.Apply(newVersion, bool(ppp == p.Spec.Priority))
 				// record the evaluated policy priority for next loop
 				ppp = p.Spec.Priority
 			}
 		}
-		found.Spec.DpConfigVersion = cksum
-		err = r.Update(context.TODO(), found)
+		newVersion.Spec.DpConfigVersion = cksum
+		if reflect.DeepEqual(found.Spec, newVersion.Spec) {
```
Comment (pliurh) [inline controllers/sriovnetworknodepolicy_controller.go:299]: Shall we also user `equality.Semantic.DeepDerivative` here?

Comment (fedepaol) [reply controllers/sriovnetworknodepolicy_controller.go:299]: I added it in the daemonset as `deepequal` wasn't working, but I guess it won't hurt / will make the code future proof.
Will change

Comment (fedepaol) [reply controllers/sriovnetworknodepolicy_controller.go]: Done

**Pull request merged**